classes/Homepage.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace MySociety\TheyWorkForYou;
+
+class Homepage {
+
+    private $db;
+
+    protected $mp_house = 1;
+    protected $mp_url = 'yourmp';
+    protected $page = 'overview';
+
+    protected $recent_types = array(
+        'DEBATELIST' => array('recent_debates', 'debatesfront', 'Commons debates'),
+        'LORDSDEBATELIST' => array('recent_debates', 'lordsdebatesfront', 'Lords debates'),
+        'WHALLLIST' => array('recent_debates', 'whallfront', 'Westminster Hall debates'),
+        'WMSLIST' => array('recent_wms', 'wmsfront', 'Written ministerial statements'),
+        'WRANSLIST' => array('recent_wrans', 'wransfront', 'Written answers'),
+        'StandingCommittee' => array('recent_pbc_debates', 'pbcfront', 'Public Bill committees')
+    );
+
+    public function __construct() {
+        $this->db = new \ParlDB;
+    }
+
+    public function display() {
+        global $this_page;
+        $this_page = $this->page;
+
+        $data = array();
+
+        $data['debates'] = $this->getDebatesData();
+        $data['mp_data'] = $this->getMP();
+        $data['regional'] = $this->getRegionalList();
+        $data['popular_searches'] = $this->getPopularSearches();
+        $data['urls'] = $this->getURLs();
+        $data['calendar'] = $this->getCalendarData();
+        $data['featured'] = $this->getEditorialContent();
+
+        return $data;
+    }
+
+    protected function getMP() {
+        $mp_url = new \URL($this->mp_url);
+        $mp_data = array();
+        global $THEUSER;
+
+        if ($THEUSER->isloggedin() && $THEUSER->postcode() != '' || $THEUSER->postcode_is_set()) {
+            // User is logged in and has a postcode, or not logged in with a cookied postcode.
+
+            // (We don't allow the user to search for a postcode if they
+            // already have one set in their prefs.)
+
+            // this is for people who have e.g. an English postcode looking at the
+            // Scottish homepage
+            try {
+                $MEMBER = new Member(array('postcode'=>$THEUSER->postcode(), 'house'=> $this->mp_house));
```

Comment:
As with the other PR, I think this won't work for Scotland page with a postcode in a place that has differing SP/UK constituency names.